### PR TITLE
Idempotent build: file-entities.php

### PR DESCRIPTION
**Pull request by alfsb**
There are two principal sources of non idempotence in PHP manuals. `scripts/file-entities.php.in` and `configure.php` This PR rewrites `scripts/file-entities.php.in` into idempotent `scripts/file-entities.php`.

`scripts/file-entities.php.in` comes a long way. It's almost as old as the first solid Docbook XML manual, and got heavily patched in its more than two decades of existence. It had and/or has code for idiosyncrasies of CVS, SVN and git, of Saxon, xsltproc and PhD, of all versions of PHP starting with PHP 4, of almost all operations systems starting with Windows 95... That is a lot of history, and that is a lot of accumulated anachronisms.

Tested on all languages with two behavior changes, listed below.
1. `<!ENTITY>` output order is changed in manual DTD preambule;
2. The commented out entities on https://github.com/php/doc-en/blob/master/reference/swoole/book.xml stop being rendered, in all languages.

The first one is insignificant, but the second one may be a real bug. It almost got me crazy because the commented out entities *are* rendered on the PHP manual main site, and the code itself hinted of some form of recursive cascading file inclusions.

But tests with PhD xhtml and bigxhtml showed that all differences are basically Swoole related, so there is no any (more?) automatic recursive file loading in `reference/` or elsewhere.

### CODE REVIEW

**Comment by alfsb** [comment]
Conflict resolved. Passes CI, and yet I would like testing of Windows users. I'm trying to make the code as shell free as possible. In particular, I would like to hear if the [`git -C`](https://github.com/php/doc-base/pull/201/files#diff-72f941dd66490c5d4d1315814f9ba8c7cd1329fce421907e5716156718527235R323) switch avoids the need for [`cd /d`](https://github.com/php/doc-base/commit/6f04778609e3bf88f4a539c2afca8494157805b1) on Windows.

The Swoole bit may warrant opening an issue, if the docs are still actively developed.

**Review by Girgias — COMMENTED** [review]
Haven't tried locally, but for Windows maybe @cmb69 could try?

File: scripts/file-entities.php [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    $test = realpath( $path );
+    if ( $test === false || is_dir( $path ) == false )
+    {
+        echo "Language directory not found: $path\n.";
+        exit(-1);
+    }
+    $path = $test;
+
+    if ( BACKPORT ) // Spurious file generated outside reference/
+        pushEntity( "language.predefined.entities.weakreference", path: "$root/$lang/language/predefined/entities.weakreference.xml" );
```
[the remark this answers is not in the document]
Comment (alfsb) [reply scripts/file-entities.php:216]: I think was a leftover or straight up bug, of a bygone era. The manual *now* goes fine without it, but while rewriting this fila there was a lot of... ghost chasing, until things started building again. Note that `BACKPORT` is already `false` and the positive branches of `BACKPORT` are TODOed to be removed.

But I cannot prove this negative in all doc- and web- repos, so it's still here for time being, to make easy to revert if it does break something, somewhere.

File: scripts/file-entities.php [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+$outfile = __DIR__ . "/../temp/file-entities.ent";
+touch( $outfile );
+$outfile = realpath( $outfile );
+
+$file = fopen( $outfile , "w" );
+if ( ! $file )
+{
+    echo "Failed to open $outfile\n.";
+    die(-1);
```
[the remark this answers is not in the document]
Comment (alfsb) [reply scripts/file-entities.php:110]: Yes. Already fixed some in `configure.php` also.

File: scripts/file-entities.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Girgias) [reply scripts/file-entities.php:216]: ACK

**Comment by cmb69** [comment]
> It almost got me crazy because the commented out entities _are_ rendered on the PHP manual main site, and the code itself hinted of some form of recursive cascading file inclusions.

Well, there may be nested comments, which are not allowed in XML.

> but for Windows maybe @cmb69 could try?

Gave it a quick try, and detected no issues.

**Comment by alfsb** [comment]
It's time. Merging.

The breakages of `&reference.spl.entities.countable;` are unrelated, as they occurs already.

**Comment by alfsb** [comment]
> Gave it a quick try, and detected no issues.

Thanks for the testing. Building manual on Windows seems to be way more viable now (compared to 2000, at least). 

https://github.com/php/doc-en/pull/4288 is a thing everyone that builds on Windows should keep a eye, as it is a nasty thing that may caused so much... suffering, for many and many years. Bit me hard, today.

**Pull request merged**